js.js

```diff
@@ -0,0 +1,286 @@
+
+
+function ChartView(data,parent,structure){
+
+  this.data = data;
+  //spread operator niewydajny bo rozbija array na liste jego elementow - używa pętli for
+  //this.max = Math.max(...this.data.map(obj => obj.units));
+  //to poniżej też używa pętli for
+  //this.max = Math.max.apply(null,this.data.map(obj => obj.units));
+  /*this.max = this.data.reduce((a,b,index)=>{
+    if(index==1){
+      return Math.max(a.units,b.units);
+    }else{
+      return Math.max(a,b.units);
+    }
+  });*/
+  this.max = this.data.map(obj => obj.units).reduce((a,b)=>Math.max(a,b));
+  this.maxColumnHeightPixels = 450;
+  this.structure = structure;
+
+
+//CREATE CHART DIV
+  this.chart = document.createElement("div");
+  this.chart.style.width = "600px";
+  this.chart.style.display = "flex";
+  this.chart.style.flexFlow = "row wrap";
+
+//CREATE BACKGROUND
+  this.background = document.createElement("div");
+  this.background.style.height = "450px";
+  this.background.style.width = "450px";
+  this.background.style.display = "flex";
+  this.background.style.flexFlow = "row nowrap";
+  this.background.style.alignItems = "flex-end";
+
+//CREATE DATE AXIS
+  this.dateAxis = document.createElement("div");
+  this.dateAxis.style.height = "20px";
+  this.dateAxis.style.width = "451px";//+unit axis width
+  this.dateAxis.style.display = "flex";
+  this.dateAxis.style.flexFlow = "row nowrap";
+  this.dateAxis.style.justifyContent = "flex-end";
+  this.dateAxis.style.marginTop = "10px";
+
+//CREATE UNIT AXIS
+  this.unitAxis = document.createElement("div");
+  this.unitAxis.style.display = "flex";
+  this.unitAxis.style.flexFlow = "column-reverse nowrap";
+  this.unitAxis.style.height = "450px";
+  this.unitAxis.style.width = "1px";
+  this.unitAxis.style.alignItems = "center";
+
+
+//RENDERING
+  this.createColumns(this.structure);
```

Comment:
ekstra metody - bardzo czytelne ... tylko one mutuja Ci (co nie jest zle) ale jestes tutaj troche nispojny ... bo w srodku tych metod uzywasz pól takich jak `background ` albo `dataAxis` a `structure` przekazujesz w parametrze .

Zeby byc spojnym to albo powinines nic nie przyjmowac w parametrze i w srodku metody brac dane z `this`
Albo tak jak teraz masz tylko zwracac DOM node i appendowac go na zewnatrz (tak zeby w srodku metody nie uzywac `this`)